modules/navier_stokes/doc/content/source/problems/NavierStokesProblem.md

```diff
@@ -0,0 +1,97 @@
+# NavierStokesProblem
+
+The `NavierStokesProblem` solves the saddle point incompressible Navier-Stokes
+equations using Schur complement field split preconditioning. It focuses on the
+Least Squares Commutator (LSC) preconditioner, also known as the BFBt
+preconditioner, developed by Elman in [!citep](elman1999preconditioning) and
+[!citep](elman2006block). The PETSc manual page on the LSC preconditioner can be
+found [here](https://petsc.org/release/manualpages/PC/PCLSC/).
+
+A linearized version of the incompressible Navier-Stokes equations discretized
+with stable finite element pairings may be written as:
+
+\begin{equation}
+\begin{bmatrix}
+A & B^T\\
+B & 0
+\end{bmatrix}
+\begin{bmatrix}
+u\\
+p
+\end{bmatrix}
+=
+\begin{bmatrix}
+f\\
+0
+\end{bmatrix}
+\end{equation}
+
+where $u$ denotes velocity degrees of freedom, $p$ denotes pressure degrees of
+freedom, and $f$ denotes the effect of body forces not dependent on the
+velocity/pressure degrees of freedom (e.g. gravity). The LSC preconditioner
+proposed by Elman approximates the Schur complement inverse via
+
+\begin{equation}
+\hat{S}^{-1} := (B\hat{M}_u^{-1}B^T)^{-1}B\hat{M}_u^{-1}A\hat{M}_u^{-1}B_T(B\hat{M}_u^{-1}B_T)^{-1}
+\end{equation}
+
+where $\hat{M}_u$ is a diagonal approximation of the velocity mass matrix. In
+the case of continuous pressure elements (per [!citep](olshanskii2007pressure)),
+the above can be re-written as
+
+\begin{equation}
+\hat{S}^{-1} := L_p^{-1}B\hat{M}_u^{-1}A\hat{M}_u^{-1}B^TL_p^{-1}
+\end{equation}
+
+where $L_p$ represents a Poisson-like operator with dimension corresponding to
+the number of pressure degrees of freedom, hence the $p$ subscript. PETSc has
+full support for LSC preconditioning. The PETSc user may optionally provide two
+"auxiliary" matrices for LSC preconditioning, corresponding to $L_p$ and
+$\hat{M}_u$, which correspond to the `NavierStokesProblem` parameters
+[!param](/Problem/NavierStokesProblem/L_matrix) and
+[!param](/Problem/NavierStokesProblem/mass_matrix) respectively. Note that in
+order to have $\hat{M}_u^{-1}$ scaling, the PETSc option `-pc_lsc_scale_diag`
+must be supplied. If `-pc_lsc_scale_diag` is not provided, then $\hat{M}_u$ is
+implicitly the identity vector. If `-pc_lsc_scale_diag` is supplied and
+[!param](/Problem/NavierStokesProblem/mass_matrix) is not provided, then
+$\hat{M}_u$ will be populated with the diagonal of $A$. If
+[!param](/Problem/NavierStokesProblem/L_matrix) is not provided, then it will be
+computed automatically via
```

Comment:
from your litterature study, can you make recommendations on these parameters based on the flow regime?

alternatively, can you point the references you used to the selection of each parameter?

For example for commute_lsc,I know to go read the Olshankii paper to under why I would use that

Reply:
I've added a section at the bottom of this file offering guidance on preconditioner selection